wasm2c/wasm-rt-impl.c

```diff
@@ -130,19 +134,25 @@ void wasm_rt_allocate_memory(wasm_rt_memory_t* memory,
 
   /* Reserve 8GiB. */
   void* addr =
-      mmap(NULL, 0x200000000ul, PROT_NONE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
+      os_mmap_aligned(NULL, 0x200000000ul, MMAP_PROT_NONE, MMAP_MAP_NONE,
+                      WASM_HEAP_ALIGNMENT, /*alignment_offset=*/0);
```

Comment:
is `os_mmap_aligned` ever called with a non-zero `alignment_offset`? If not, then maybe that extra code should be removed.